### ISSUE TITLE
BooleanField does not render properly with FormField and FieldList

### ISSUE THREAD

**Issue by Amertz08** [wtforms/wtforms#308]
I am creating a dynamic length form based on multiple db entries and I cannot get it to override the default checked value for the BooleanField.

```python
class PackageForm(Form):
    pack_id = HiddenField('Pack ID')
    desc = HiddenField('Description')
    mrr = HiddenField('MRR')
    include = BooleanField('Include', default=True)

class PackageListForm(Form):
    packs = FieldList(FormField(PackageForm))
    submit = SubmitField('Submit')

# In view
services = Service.query.all()

form_fields = [
    PackageForm(
        pack_id=service.pack_id,
        desc=service.description,
        mrr=service.mrr,
        include=service.include
        ) for service in services
]

for field in form_fields:
    print(field.include)

form = PackageListForm(packs=form_fields)
```
And the print statement prints the correct html. But when it gets rendered it makes all field checked.
```html
<!--from print statement -->
<input id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
<input checked id="include" name="include" type="checkbox" value="y">
```
Haven't been able to find a work around for it yet.
Here is the template.
```html
<form action="" method="post" class="form" role="form">
    {{ form.csrf_token }}
    <table class="table table-bordered table-striped">
        <thead>
        <tr>
            <th>Pack ID</th>
            <th>Description</th>
            <th>MRR</th>
            <th>Include</th>
        </tr>
        </thead>
        <tbody>
        {% for field in form.packs %}
        <tr>
            <td>{{ field.form.csrf_token }}
                {{ field.form.pack_id }}{{ format_form(field.form.pack_id.data) }}</td>
            <td>{{ field.form.desc }}{{ format_form(field.form.desc.data) }}</td>
            <td>{{ field.form.mrr }}{{ format_form(field.form.mrr.data) | dollar }}</td>
            <td><label>{{ field.form.include }}</label></td>
        </tr>
        {% endfor %}
        </tbody>
    </table>
    <input type="submit" class="btn btn-default">
</form>
```

**Comment by Amertz08** [comment]
```python
class BooleanSubField(BooleanField):

    def process_data(self, value):
        if isinstance(value, BooleanField):
            self.data = value.data
        else:
            self.data = bool(value)
```

So I think I figured out the issue and a possible resolution. When you use FormField with a form that has a BooleanField the process_data function on the BooleanField gets called twice. One of those times value gets passed as an instance of a BooleanField and not a boolean built in. Because of this `bool(value)` will always return true since it is simply being called on an object. I ended up doing the above for my project.
